cartographer_ros/cartographer_ros/map_builder_bridge.cc

```diff
@@ -196,4 +196,34 @@ SensorBridge* MapBuilderBridge::sensor_bridge(const int trajectory_id) {
   return sensor_bridges_.at(trajectory_id).get();
 }
 
+visualization_msgs::MarkerArray MapBuilderBridge::GetTrajectoryNodesList() {
+  visualization_msgs::MarkerArray trajectory_nodes_list;
+
+  for (int trajectory_id = 0;
+       trajectory_id < map_builder_.num_trajectory_builders();
```

Comment:
This loop is redundant. `GetTrajectoryNodes()` below returns the data for all trajectories. If we loop here an again in line 209, we publish each trajectory multiple times. Remove this loop?